File: assignments/a1/q9/Java/Token_key.java
//Token_key:top//
//Token_key:import//
import java.util.*;

public abstract class Token_key /*Token_key:class*/ {

    public static final String $className = "Token_key";
    public static Token_key parse(Scan scn$, Trace trace$) {
        Token t$ = scn$.cur();
        Token.Match match$ = t$.match;
        switch(match$) {
        case TOKEN_NAME:
            return TokenNoKey.parse(scn$,trace$);
        case TOKEN_KEY:
            return TokenWithKey.parse(scn$,trace$);
        default:
            throw new PLCCException(
                "Parse error",
                "Token_key cannot begin with " + t$.errString()
            );
        }
    }

//Token_key//
}

File: assignments/a1/q9/Python/TokenKey.py
#TokenKey:top#
#TokenKey:import#

# <token_key>TokenKey ::= <TOKEN_KEY> <TOKEN_NAME>
class TokenKey(Token_key): #TokenKey:class#

    className = "TokenKey"
    ruleString = "<token_key>TokenKey ::= <TOKEN_KEY> <TOKEN_NAME>"
    token_key = None
    token_name = None

    def __init__(token_key, token_name):
        #TokenKey:init#
        self.token_key = token_key
        self.token_name = token_name

#TokenKey#
